src/MongoDB.Driver.Core/Core/Operations/QueryHelper.cs

```diff
@@ -92,7 +92,8 @@ public static BsonDocument CreateReadPreferenceDocument(ReadPreference readPrefe
             {
                 { "mode", modeString },
                 { "tags", tagSets, tagSets != null },
-                { "maxStalenessSeconds", () => (int)readPreference.MaxStaleness.Value.TotalSeconds, readPreference.MaxStaleness.HasValue }
+                { "maxStalenessSeconds", () => (int)readPreference.MaxStaleness.Value.TotalSeconds, readPreference.MaxStaleness.HasValue },
```

Comment:
This method when it will be called from `CommandUsingCommandMessageWireProtocol` has completely different condition (only about _readPreference != ReadPreference.Primary, see [this](https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs#L287-L289)). 
Since the spec uses: `MUST` and `MUST NOT` can we avoid all other condition requirements?

Reply:
This is about whether the ReadPreference can be encoded using only a slaveOk bit